central/detection/lifecycle/manager_impl.go

```diff
@@ -133,6 +147,21 @@ func indicatorToBaselineKey(indicator *storage.ProcessIndicator) processBaseline
 	}
 }
 
+func (m *managerImpl) flushBaselineQueue() {
+	for {
+		// ObservationEnd is in the future so we have nothing to do at this time
+		head := m.deploymentObservationQueue.Peek()
+		if head == nil || protoutils.After(head.ObservationEnd, types.TimestampNow()) {
+			return
+		}
+
+		// Grab the first deployment to baseline.
+		deployment := m.deploymentObservationQueue.Pull()
```

Comment:
Can you add a comment saying that this is the only place pulls happen? Otherwise, this constitutes a race condition and could panic